### PR TITLE
virtctl: cloneAPI add template filters

### LINKED ISSUE #10374: Supportr "Clone API" to filter VirtualMachine.spec.template.annotation and labels
**Is your feature request related to a problem? Please describe**:
A clear and concise description of what the problem is.
we used kube-ovn as our kubevirt cluster cni. as is known to all, this kube-ovn cni will add the network's info in "virtualmachine.spec.template.annotation"
image
it will raise an issue when we use "Clone API" to clone a new vm. the clone vm would use the same network. so we want to add annotation Filter to avoid this problems.
**Describe the solution you'd like**:

**Describe alternatives you've considered**:
```yaml
kind: VirtualMachineClone
apiVersion: "clone.kubevirt.io/v1alpha1"
metadata:
  name: testclone

spec:
  # source & target definitions
  source:
    apiGroup: kubevirt.io
    kind: VirtualMachine
    name: vm-cirros
  target:
    apiGroup: kubevirt.io
    kind: VirtualMachine
    name: vm-clone-target

  # labels & annotations definitions
  labelFilters:
  - "*"
  - "!someKey/*"
  annotationFilters:
  - "anotherKey/*"
  templateFilters:
    labelFilters:
    - "*"
    - "!someKey/*"
    annotationFilters:
    - "anotherKey/*"

  # other identity stripping specs:
  newMacAddresses:
    interfaceName: "00-11-22"
  newSMBiosSerial: "new-serial"

### PR DESCRIPTION
**Pull request by matthewei**
**What this PR does / why we need it**:
virtctl support to add template label and annotation filters
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #10374 

**Special notes for your reviewer**:
https://github.com/kubevirt/kubevirt/pull/10658
**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
  # Create a manifest for a clone with template label filters:
  virtctl create clone --source-name sourceVM --template-label-filter '*' --template-label-filter '!some/key'

  # Create a manifest for a clone with template annotation filters:
  virtctl create clone --source-name sourceVM --template-annotation-filter '*' --template-annotation-filter '!some/key'

```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
Not sure I'm the biggest fan of case-matching on strings, but it looks OK.

/ok-to-test

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Comment by alicefr** [comment]
> Not sure I'm the biggest fan of case-matching on strings, but it looks OK.
> 
> /ok-to-test

I agree with Felix, you could use constants like:
```golang
const (
     label = iota
     annotations
     all
     ...
)
```

**Comment by matthewei** [comment]
> > Not sure I'm the biggest fan of case-matching on strings, but it looks OK.
> 
> I agree with Felix, you could use constants like:我同意 Felix 的观点，您可以使用以下常量：
> 
> ```go
> const (
>      label = iota
>      annotations
>      all
>      ...
> )
> ```

ok, I will modify it.

File: pkg/virtctl/create/clone/clone_test.go
Comment (fossedihelm) [inline pkg/virtctl/create/clone/clone_test.go:47]: These can be unexported
```suggestion
const (
	labelFilters = iota
	annotationsFilters
	labelAndAnnotationsFilters
	templateLabelFilters
	templateAnnotationsFilters
	templateLabelAndAnnotationsFilters
)
```

Thanks

**Comment by matthewei** [comment]
Thanks, all let's move to next step.

**Comment by matthewei** [comment]
Do we need this function?

**Comment by 0xFelix** [comment]
Ping @alicefr 

Would you mind taking a look again?

**Comment by matthewei** [comment]
@fossedihelm could you have a look at is patch ?

**Comment by fossedihelm** [comment]
Hey @matthewei I have already given my lgtm to this. Let's wait for @alicefr response. These are holiday days somewhere, so you may experience a delay in replying :)

**Comment by alicefr** [comment]
This change looks good to me
/approve

**Pull request merged**
